File: tests/python/src/pages/base_page.py
class BasePage:
    """Generic base page with helper methods"""

    def __init__(self, page):
        self.page = page

    def wait_for_page_load(self):
        """Wait for the network to be idle. Override in subclasses if necessary."""
        self.page.wait_for_load_state("load")

    def is_displayed(self) -> bool:
        return self.page.is_visible("body") 

File: tests/typescript/src/pages/base.page.ts
import { Page } from '@playwright/test'

export class BasePage {
  protected readonly page: Page

  constructor(page: Page) {
    this.page = page
  }

  // Wait for generic page load; subclasses should override for specific conditions
  async waitForPageLoad(): Promise<void> {
    await this.page.waitForLoadState()
  }

  // Basic visibility check; subclasses can override
  async isDisplayed(): Promise<boolean> {
    return this.page.isVisible('body')
  }
} 